app/forms/validators.py

```diff
@@ -424,9 +449,20 @@ def __call__(
         is_valid, message = self._is_valid(condition, total, target_total)
 
         if not is_valid:
+            decimal_places = (
+                2
+                if isinstance(target_total, int)
```

Comment:
Why do we default to 2 here?.

Reply:
Have reverted this temporarily, there is an issue in `_get_target_total_and_currency` which i think is always returning an int when calculating the target total. So that method needs to be fixed

Reply:
The total should return based on no. of places we need to display.

Reply:
Yeah I get the logic, like it's just that that method at the moment doesn't return a decimal so at the moment it is not possible to figure out the correct number of decimals to be displayed. So looks like that `_get_target_total_and_currency` needs to be refactored

Reply:
Is there a lot to refactoring?. Don't you just need to return the max decimal place for the answers to calculate?.
It would be done in `_validate_calculated_question` since that is where the `SumCheck` validator is called, and you have calculation and question in scope.

Same thing applies to calc/grand calc summaries. I think it is an issue there also, as it currently get the answer format for the first answer. Do we have any validation to ensure all answers must have same no. of decimal places?.

Reply:
No there is nothing stopping answers having different decimal limits which are then being used in calc summaries etc, added a schema for this too 👍

Reply:
Added a fix for the initial issue now 👍

Reply:
Great stuff.